### PR TITLE
API Gateway integration tests refactoring

### PR DESCRIPTION
**Pull request by cnvergence**
**Description**

Changes proposed in this pull request:

- API Gateway tests are triggered on component changes using minimal setup.
- Run on gardener cluster
- Moving away from Octopus testing

**Related issue(s)**

Resolves [#11323](https://github.com/kyma-project/kyma/issues/11323)

### CODE REVIEW

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  - name: "dex"
+  - name: "ory"
+  - name: "api-gateway"
+EOF
+}
+
+# uses previously set KYMA_SOURCE
+if [[ "${KYMA_ALPHA}" == "true" ]]; then
+  host::prepare_components_file
+  log::info "Installing Kyma with Dex, Ory and API-Gateway"
```
Comment (mjakobczyk) [inline prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:138]: We log the same message in both cases. Shouldn't we log it once?

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  kyma alpha deploy --ci --components-file "$PWD/components.yaml" --value global.isBEBEnabled=true --source=local --workspace "${KYMA_SOURCES_DIR}" --verbose
+else
+  host::prepare_components_file
+  log::info "Installing Kyma with Dex, Ory and API-Gateway"
```
Comment (mjakobczyk) [inline prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:142]: We log the same message in both cases. Shouldn't we log it once?

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+cat << EOF > "$PWD/components.yaml"
+defaultNamespace: kyma-system
+prerequisites:
+  - name: "cluster-essentials"
+  - name: "istio"
+    namespace: "istio-system"
+  - name: "certificates"
+    namespace: "istio-system"
+components:
+  - name: "dex"
```
Comment (mjakobczyk) [inline prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:129]: What is the purpose of using Dex in `kyma alpha deploy` scenario?

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 4]
```diff
@@ -240,6 +240,54 @@ presubmits: # runs on PRs
...
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pre-main-kyma-gardener-gcp-api-gateway"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-build-pr: "true"
+        preset-cluster-version: "true"
+        preset-debug-commando-oom: "true"
+        preset-gardener-gcp-kyma-integration: "true"
+        preset-kyma-cli-stable: "true"
+      optional: true
```
Comment (strekm) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:253]: will there be follow up PR to make it mandatory?

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 5]
```diff
@@ -240,6 +240,54 @@ presubmits: # runs on PRs
...
+      decorate: true
+      decoration_config:
+        grace_period: 600000000000
+        run_if_changed: ^(((resources/api-gateway)\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))
+        timeout: 14400000000000
+      path_alias: github.com/kyma-project/kyma
+      cluster: untrusted-workload
+      max_concurrency: 10
+      branches:
+        - ^master$
```
Comment (strekm) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:264]: i think this one is not necessary anymore

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 6]
```diff
@@ -240,6 +240,54 @@ presubmits: # runs on PRs
...
+        - ^main$
+      extra_refs:
+        - org: kyma-project
+          repo: test-infra
+          path_alias: github.com/kyma-project/test-infra
+          base_ref: main
+      spec:
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/kyma-integration:v20210519-51c0242d-k8s1.16"
```
Comment (strekm) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:273]: is that the latest image?

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 7]
```diff
@@ -240,6 +240,54 @@ presubmits: # runs on PRs
               requests:
                 memory: 1Gi
                 cpu: 400m
+    - name: pre-main-kyma-gardener-gcp-api-gateway
```
Comment (strekm) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:243]: maybe adding post master job as well?

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh (hunk 3 above)
Comment (strekm) [inline prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:129]: long term we need to think how to replace dex

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set KYMA_SOURCE used by gardener::install_kyma
```
Comment (strekm) [inline prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:81]: is it used by `kyma alpha deploy`?

File: prow/jobs/kyma/kyma-integration-gardener.yaml (hunk 4 above)
Comment (cnvergence) [reply prow/jobs/kyma/kyma-integration-gardener.yaml:281]: the prow/jobs are optional by default, so the render templates tool will change this setting when executed

File: prow/jobs/kyma/kyma-integration-gardener.yaml (hunk 5 above)
Comment (cnvergence) [reply prow/jobs/kyma/kyma-integration-gardener.yaml:292]: this also added by default by the render templates tool

File: prow/jobs/kyma/kyma-integration-gardener.yaml (hunk 6 above)
Comment (cnvergence) [reply prow/jobs/kyma/kyma-integration-gardener.yaml:301]: yes, it is

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh (hunk 3 above)
Comment (cnvergence) [reply prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:104]: Current tests are dependent on Dex, admin-user is used in the tests.

Comment (cnvergence) [reply prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh:104]: 👍  we will need to redesign the tests, currently they are dependent on admin-user

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh (hunk 1 above)
Comment (cnvergence) [reply prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh]: 👍

File: prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh (hunk 8 above)
Comment (cnvergence) [reply prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh]: no indeed, I will remove that leftover from there

File: prow/jobs/kyma/kyma-integration-gardener.yaml (hunk 7 above)
Comment (cnvergence) [reply prow/jobs/kyma/kyma-integration-gardener.yaml:271]: 👍  let's do it

File: prow/jobs/kyma/kyma-integration-gardener.yaml (hunk 4 above)
Comment (strekm) [reply prow/jobs/kyma/kyma-integration-gardener.yaml:281]: yes ofc i meant to change template in a way this job will be mandatory

**Review by strekm — DISMISSED** [review]
/meow

**Review by mjakobczyk — APPROVED** [review]

**Comment by cnvergence** [comment]
Stale review was dismissed due to review from prow team, together with @adamwalach we wanted to reduce additional maintenance costs of new gardener-integration scripts with every component.
We will use the standard kyma_gardener_integration.sh extended by integration_tests.sh helper and another component tests will further extend it with new environment variables/

**Review by adamwalach — APPROVED** [review]

**Pull request merged**